src/sage/combinat/SJT.py

```diff
@@ -0,0 +1,141 @@
+r"""
+Steinhaus-Johnson-Trotter algorithm.
+
+The Steinhaus-Johnson-Trotter algorithm generates permutations in a specific
+order by transposing only two elements from the list at each operation. The
+algorithm stores an internal state for every element in the permutated list
+which corresponds to their direction. To know which elements to move, the
+internal state table is accessed and the transposition of two elements is then
+done.
+
+It is important to notice that the permutations are generated in a different
+order than the default lexicographic algorithm.
+
+The class defined here is meant to be used in the ``Permutation`` class when
+called with the parameter ``algorithm='sjt'``.
+
+AUTHORS:
+
+- Martin Grenouilloux (2024-05-22): initial version
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2024 Martin Grenouilloux <martin.grenouilloux@uib.no>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+class SJT:
```

Comment:
You need a class string. An approximate version of the string at the top of the file would work. It should just be an explanation of how the class works, along with examples i.e. doctests of usage.